### ISSUE TITLE
Allow '/' character for types.identifier()

### ISSUE THREAD

**Issue by leesiongchan** [mobxjs/mobx-state-tree#88]
In our case, we use [IRI](https://en.wikipedia.org/wiki/Internationalized_Resource_Identifier) as our `id`, do you think it possible to allow `/` to be used in `types.identifier()`? Or it will break the reference path? Or any other reasons you make it strict for alphanumeric only?

Here is my sample user
```javascript
const imageFromApi = {
  id: '/images/1',
  contentUrl: 'https://upload.wikimedia.org/wikipedia/commons/thumb/b/bc/Leonardo_DiCaprio_January_2014.jpg/212px-Leonardo_DiCaprio_January_2014.jpg',
};
const userFromApi = {
  id: '/users/1',
  image: '/images/1',
  name: 'Leonardo DiCaprio',
};
```

**Comment by mattiamanzati** [comment]
@leesiongchan Uhm, it should'nt break anything, identifier is already mapped to a string, and it's contents can be literaly anything :)

**Comment by mweststrate** [comment]
nonetheless, feel free to add additional test-cases to the test suite if you want to test the behavior of mobx with weird identifiers! (tests can be written in both js and ts, whatever suits you)

**Comment by mattiamanzati** [comment]
Should be fixed once #152 gets merged! :)
